pwndbg/commands/asm.py

```diff
@@ -0,0 +1,74 @@
+import argparse
+
+import pwn
+import pwnlib.context
+
+import pwndbg.commands
+
+parser = argparse.ArgumentParser(
+    description="Runs cpp() over a given shellcode and then assembles it into bytes"
+)
+
+parser.add_argument(
+    "-f", "--format", default="hex", choices=["hex", "string"], type=str, help="Output format"
+)
+
+parser.add_argument(
+    "--arch",
+    default=pwnlib.context.context.arch,
+    choices=pwnlib.context.context.architectures.keys(),
+    type=str,
+    help="Target architecture",
+)
+
+parser.add_argument(
+    "-v",
+    "--avoid",
+    action="append",
+    help="Encode the shellcode to avoid the listed bytes (provided as hex)",
+)
+
+parser.add_argument(
+    "-n",
+    "--newline",
+    dest="avoid",
+    action="append_const",
+    const="0a",
+    help="Encode the shellcode to avoid newlines",
+)
+
+parser.add_argument(
+    "-z",
+    "--zero",
+    dest="avoid",
+    action="append_const",
+    const="00",
+    help="Encode the shellcode to avoid NULL bytes",
+)
+
+input_group = parser.add_mutually_exclusive_group(required=True)
+
+input_group.add_argument(
+    "shellcode", default=[], nargs="*", type=str, help="Assembler code to assemble"
+)
+
+input_group.add_argument("-i", "--infile", default=None, type=str, help="Specify input file")
+
+
+@pwndbg.commands.ArgparsedCommand(parser, command_name="asm")
+def asm(shellcode, format, arch, avoid, infile):
+    if infile:
+        with open(infile) as file:
```

Comment:
can we have a warning print if shellcode is set to anything that it will be ignored and instead the asm will be read from infile?